hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestChecker.java

```diff
@@ -142,6 +148,20 @@ public TransactionInfo runAllChecks(@NonNull final HederaState state, @NonNull f
         return txInfo;
     }
 
+    public void checkDuplicates(final TransactionBody txBody) throws PreCheckException {
+        // TODO: not sure if this is how it should be implemented
+        // We are throwing an error only if the transactionId is in the cache and if the
+        // submitter is the creator of the transaction
+        final var foundTransactionRecord = hederaRecordCache.getRecord(txBody.transactionIDOrThrow());
+        if (foundTransactionRecord != null) {
+            if (txBody.nodeAccountID() != null
+                    && txBody.nodeAccountID()
+                            .equals(txBody.transactionIDOrThrow().accountID())) {
+                throw new PreCheckException(DUPLICATE_TRANSACTION);
+            }
+        }
+    }
+
```

Comment:
I think actually in ingest checker we should just remove the "step 2" TODO. We do deduplication check in SubmissionManager and don't need to do a check here.